CDCgov/prime-reportstream
# Support FHIR receivers and receiver transforms

This PR adds all of the necessary functionality to allow for receivers to be created in the FULL_ELR topic with a FHIR format to receive FHIR messages. This involved a lot of pieces to make things work seamlessly, which I will break down below in the `Changes` section.

Test Steps:
1. Reload the settings with `./gradlew reloadSettings`
2. Send a report with a state of `IG` to trigger the ignore receivers
3. Check the sftp folder for resulting batched messages; you should see one in HL7 format and one in FHIR format

## Changes
- Added a placeholder receiver transform to verify that receiver fhir transforms are applied
- Added a new FHIR receiver to the ignore organization and update smoke tests to account for that (including addition of `TestReportStream.getAllChildrenReportId`
- Added `useBatching` getter to `Receiver` to determine whether or not to batch messages for a receiver
- Moved `HL7MessageHelpers.takeHL7GetReport` to `Report.generateReportAndUploadBlob` (with basically no changes)
- Added `FHIRConfiguration` subtype of `TranslatorConfiguration` and add `useBatching` getter to `TranslatorConfiguration` (used in conjunction with the same getter on `Receiver`)
- Added `FHIRBundleHelpers.batchMessages` to batch FHIR messages into a single `fhir+ndjson` file
- Updated the default processing mode filter to prevent exceptions
- Refactored `FHIRTranslator` to allow for
  - flexibility to translate to HL7v2 or FHIR
  - clean pruning of endpoints/conditions that shouldn't be sent to a given receiver
- Added/adjusted tests

## Checklist

### Testing

## Linked Issues
- Addressed #9144
